src/Rules/AllOfRule.php

```diff
@@ -39,13 +39,13 @@ public function processValue($value, Args $args, FieldContext $context)
 					$context,
 				);
 			} catch (ValueDoesNotMatch | InvalidData $exception) {
-				$type->overwriteInvalidSubtype($key, $exception->getInvalidType());
+				$type->overwriteInvalidSubtype($key, $exception);
 				$anyValidationFailed = true;
 			}
 		}
 
 		if ($anyValidationFailed) {
-			throw ValueDoesNotMatch::create($type);
+			throw ValueDoesNotMatch::create($type, $value);
```

Comment:
$value may be already mutated here, original value from function input should be added. Test for that case is welcome

Reply:
TODO: test case

Reply:
Test case added